src/server/project.ts

```diff
@@ -2164,6 +2164,24 @@ namespace ts.server {
             this.externalProjectRefCount--;
         }
 
+        /* @internal */
+        isSolution() {
+            return this.getRootFilesMap().size === 0 &&
+                !this.canConfigFileJsonReportNoInputFiles;
+        }
+
+        /* @internal */
+        getDefaultChildProjectFromSolution(info: ScriptInfo) {

```

Comment:
>getDefaultChildProjectFromSolution [](start = 8, length = 34)

I'm not sure I understand what this does.  I think it returns the first loaded (?) project (in tsconfig project-reference-list order?) that contains the ScriptInfo directly (i.e. not because it was pulled in from a referenced project?).

Reply:
Does the answer change as more projects are loaded?

---
In reply to: [388677179](https://github.com/microsoft/TypeScript/pull/37239#discussion_r388677179) [](ancestors = 388677179)

Reply:
This tries to find the project that is a referenced project from the solution where the script info belongs (and not just referenced) eg. from project `src\tsconfig.json` for file `src\compiler.ts` it will find `src\compiler\tsconfig.json` and not `src\services\tsconfig.json`

Reply:
And does the answer depend on which projects are loaded when the question is asked?

Reply:
Yes. Because it only finds the projects